src/main/java/edu/kit/compiler/codegen/BasicBlocks.java

```diff
@@ -71,14 +72,14 @@ public int getLabel() {
          * Return the basic instructions of this block.
          */
         public List<Instruction> getInstructions() {
-            return Collections.unmodifiableList(instructions);
+            return instructions;
```

Comment:
There is a reason why this originally returned an unmodifiable list. You're not really supposed to change the state of this class. For example, I'm currently considering breaking `instructions` into two lists. You're change here makes that pretty much impossbile.